server/controllers/excelController.js

```diff
@@ -44,7 +47,100 @@ excelController.read = async (req, res, next) => {
 excelController.convertInputs = async (req, res, next) => {
   console.log('entering excelController.convertInputs');
   try {
+    const sample = { People: 'A', Species: 'I', Movies: 'Q' };
     const inputCols = {};
+    let data;
+    if (req.body.document) data = JSON.parse(req.body.document);
+
+    const arrOfColumnLetters = [];
+    const arrOfColumnNames = [];
+    // Gets column names
+    for (const column in res.locals.inputRows[0]) {
+      arrOfColumnNames.push(column);
+    }
+
+    // Gets column letters and assign columns names to letters
+    const objColumnLettersToNames = {};
+    for (const column in res.locals.raw[res.locals.targetSheet]) {
+      if (column !== '!ref' && column !== '!merges!') {
+        let str = '';
+        for (let i = 0; i < column.length; i++) {
+          if (!+column[i]) {
+            str += column[i];
+            objColumnLettersToNames[
+              res.locals.raw[res.locals.targetSheet][column].w
+            ] = str;
+          }
+        }
+        arrOfColumnLetters.push(str);
+      }
+      if (arrOfColumnLetters.length === arrOfColumnNames.length) break;
+    }
+
+    // Gets table boundaries
+    const excelMap = { ...(data || sample) };
+    const arrOfColLettersSample = Object.values(data || sample);
+    const arrOfTableNames = Object.keys(data || sample);
```

Comment:
```suggestion
    const expectedData = data || sample;
    const excelMap = { ...expectedData };
    const arrOfColLettersSample = Object.values(expectedData);
    const arrOfTableNames = Object.keys(expectedData);
```

Reply:
Pushed new changes that reflect your suggested code